vm/src/stdlib/json.rs

```diff
@@ -0,0 +1,199 @@
+use std::collections::HashMap;
+use std::fmt;
+
+use serde;
+use serde::de::Visitor;
+use serde::ser::{SerializeMap, SerializeSeq};
+use serde_json;
+
+use super::super::pyobject::{
+    DictProtocol, PyContext, PyFuncArgs, PyObject, PyObjectKind, PyObjectRef, PyResult,
+};
+use super::super::VirtualMachine;
+
+impl serde::Serialize for PyObjectKind {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self {
+            PyObjectKind::String { value } => serializer.serialize_str(value),
```

Comment:
Instead of using ObjectKind, use isinstance or alike to check for the type. This allows for subclasses of str, like `class MyStr(str)` to work as well.